react-playground/src/components/Header/Header.tsx

```diff
@@ -1,51 +1,47 @@
-import { FC, useState } from "react";
+import { FC, useState, useLayoutEffect, useRef, useCallback } from "react";
 import { Link } from "react-router-dom";
 import MainLogo from "../../images/main-logo.svg";
 import MenuIcon from "../../images/menu-icon.svg";
-import { HeaderStyled, DropdownStyled, MenuItemStyled } from "./Header.styles";
-
-const menu = [
-  { name: "Home", route: "/" },
-  { name: "My Results", route: "/results" },
-  { name: "Health profile", route: "/healthprofile" },
-  { name: "Community", route: "/community" },
-  { name: "Resources", route: "/resources" },
-  { name: "Settings", route: "/settings" },
-  { name: "Log out", route: "/logout" },
-];
+import { HeaderStyled, MenuIconStyled } from "./Header.styles";
+import { Menu } from "../Menu";
 
 const Header: FC = () => {
   const [isOpen, setIsOpen] = useState(false);
   const toggling = () => setIsOpen(!isOpen);
 
-  const onItemClicked = (value: Object) => () => {
+  const onItemClicked = () => {
     setIsOpen(false);
   };
 
+  const element = useRef<HTMLDivElement>(null);
+
+  const handleClick = useCallback((e: MouseEvent) => {
+    if (element.current && element.current.contains(e.target as HTMLElement)) {
+      return;
+    }
+    setIsOpen(false);
+  }, []);
+
+  useLayoutEffect(() => {
+    document.addEventListener("click", handleClick);
+    return () => {
+      document.removeEventListener("click", handleClick);
+    };
+  }, [handleClick]);
+
   return (
```

Comment:
I like that you got it working from our pairing session but I wonder if having the eventListener inside the useLayoutEffect is the best practice in React components - I'd love to get Hugh's and/or Jen's FE expert advice on this.

Reply:
This is pretty typical to have the add and remove as part of a useEffect; I added some notes on this below!